wrapper/src/main/java/software/amazon/jdbc/wrapper/ConnectionWrapper.java

```diff
@@ -99,10 +103,19 @@ public ConnectionWrapper(
             effectiveConnectionProvider,
             this,
             telemetryFactory);
+    ServiceContainer serviceContainer = new ServiceContainerImpl(storageService, pluginManager, telemetryFactory);
```

Comment:
I don't love how we partially initialize the ServiceContainer here and then complete the initialization with the setters below. I wonder if we should consider a proper service locator to avoid this.

Reply:
There's nothing wrong about partially initialize the object. By design this object is supposed to hold service references. Some of them could be missed, i.e. not yet initialized. I think we 're fine. We could improve it later.